index.ts

```diff
@@ -76,23 +98,94 @@ export const sandbox = {
-      (inv['tables-streams'] ?? []).map(({ table }) =>
-        dynamodbClient.send(
-          new UpdateTableCommand({
-            TableName: client.name(table),
-            StreamSpecification: {
-              StreamEnabled: true,
-              StreamViewType: 'NEW_AND_OLD_IMAGES',
-            },
-          })
+    const tableStreams: TableStreamItem[] = inv['tables-streams']
+    abortController = new AbortController()
+    if (tableStreams?.length) {
+      const ddbStreamsClient = new DynamoDBStreamsClient({
+        region: inv.aws.region,
+        endpoint: local.url,
+        credentials,
+      })
+      // Init table streams for those defined
+      await Promise.all(
+        tableStreams
+          .map(({ table }) => table)
+          .filter(
+            (item: string, index: number, tableNames: string[]) =>
+              tableNames.indexOf(item) === index
+          )
+          .map((table: string) =>
+            dynamodbClient.send(
+              new UpdateTableCommand({
+                TableName: client.name(table),
+                StreamSpecification: {
+                  StreamEnabled: true,
+                  StreamViewType: 'NEW_AND_OLD_IMAGES',
+                },
+              })
+            )
+          )
+      )
+      // Reset Stream defaults
+      for (const arcStream of tableStreams) {
+        shardMap[arcStream.table] = []
+        await resetTableStreams(
+          dynamodbClient,
+          ddbStreamsClient,
+          arcStream.table
         )
+      }
+
+      dynamoDbStreamsLoop = periodically(
+        () => streamLoop(ddbStreamsClient),
+        2000,
+        abortController.signal
       )
-    )
+    }
+
+    async function streamLoop(ddbStreamsClient: DynamoDBStreamsClient) {
+      for (const key of Object.keys(shardMap)) {
+        if (shardMap[key].length) {
+          const shardItem = shardMap[key].pop()
+          if (!shardItem) continue
+          try {
+            const event = await ddbStreamsClient.send(
+              new GetRecordsCommand({
+                ShardIterator: shardItem.ShardIterator,
+              })
+            )
+            if (event.Records?.length) {
+              tableStreams
+                .filter((x) => x.table === key)
+                .forEach((x) => {
+                  invoke({
+                    pragma: 'tables-streams',
+                    name: x.name,
+                    payload: event,
+                  })
+                })
+            }
+
+            if (event.NextShardIterator) {
+              shardMap[key].push({
+                ShardIterator: event.NextShardIterator,
+              })
+            }
+          } catch (error) {
+            if (error instanceof TrimmedDataAccessException) {
+              console.log(error.name)
+              await resetTableStreams(dynamodbClient, ddbStreamsClient, key)
+            } else {
+              throw error
+            }
+          }
+        }
+      }
+    }
```

Comment:
- You are processing at most one shard per table in this function. Probably you should consume all shards.
- You are processing shards in a last-in, first-out order (note calls to push and pop above). Surely you should process them in a first-in, first-out order.

I wonder if fixing those two issues might address the `TrimmedDataAccessException` errors.